exporter/elasticsearchexporter/exporter_test.go

```diff
@@ -82,6 +82,72 @@ func TestExporterLogs(t *testing.T) {
 		rec.WaitItems(1)
 	})
 
+	t.Run("publish with bodymap encoding", func(t *testing.T) {
+		tableTests := []struct {
+			name     string
+			body     func() pcommon.Map
+			expected string
+		}{
+			{
+				name: "flat",
+				body: func() pcommon.Map {
+					body := pcommon.NewMap()
+					body.PutStr("@timestamp", "2024-03-12T20:00:41.123456789Z")
+					body.PutInt("id", 1)
+					body.PutStr("key", "value")
+					return body
+				},
+				expected: `{"@timestamp":"2024-03-12T20:00:41.123456789Z","id":1,"key":"value"}`,
+			},
+			{
+				name: "dot",
+				body: func() pcommon.Map {
+					body := pcommon.NewMap()
+					body.PutInt("a", 1)
+					body.PutInt("a.b", 2)
+					body.PutInt("a.b.c", 3)
+					return body
+				},
+				expected: `{"a":{"value":1,"b":{"value":2,"c":3}}}`,
```

Comment:
I'm surprised that this is not producing `{"a":1,"a.b":2,"a.b.c":3}`. Why is this automatically expanding dots to a JSON object? This seems to go against the goal of this mode to just send the body to ES verbatim.

Reply:
This is exactly why I asked for this test case. OTel mode is affected by this too. It is the es exporter dedup in action. I'm afraid we'll have to disable dedup, at least for this bodymap mode. On the other hand, we'll likely have to do the same for OTel mode in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35651

Reply:
I think there is dedup and dedot taking place, I believe we shouldn't have neither enabled with this mapping mode.

I'm having a bit of trouble figuring out what it should do in this particular case:

```
body := pcommon.NewMap()
body.PutInt("a", 1)
b := body.PutEmptyMap("b")
b.PutInt("c", 2)
```

Should it serialize to `{"a":1,"b":{"c":2}}`? Even with dedup and dedot disable it is producing `{"a":1,"b.c":2}`.

Reply:
I think this case is because keys are [flattened](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/afe87cd35831e5a47b154d3a07d32c50372d3f2a/exporter/elasticsearchexporter/internal/objmodel/objmodel.go#L595C1-L600C2)  in objmodel, this causes nested maps to have dots in the key like in the example I shared above. Is this expected? I cannot find any documentation on how this should be encoded in cases of nesting in the map body.

Reply:
I might have to use a different approach here, like what the jsonlogencodingextension does with [AsRaw](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/afe87cd35831e5a47b154d3a07d32c50372d3f2a/extension/encoding/jsonlogencodingextension/extension.go#L36). I just tested it and it seems to behave as intended.

Reply:
So I decided to change the approach an bypass the objmodel call. Since we need to convert the log record body as-is I don't think we need to handle dedot, dedup and the objmodel at all. We should be able to simply `json.Marshal(record.Body().AsRaw())` and get a valid json document out. I decided to build it this way, let me know if you disagree. Also added additional test cases that were missing, for example for a body containing a slice and an inner map inside.